crates/services/consensus_module/poa/src/sync.rs

```diff
@@ -0,0 +1,594 @@
+    inner_state: InnerSyncState,
+    timer: DeadlineClock,
+}
+
+impl SyncTask {
+    pub fn new(
+        peer_connections_stream: BoxStream<usize>,
+        min_connected_reserved_peers: usize,
+        time_until_synced: Duration,
+        block_stream: BoxStream<BlockImportInfo>,
+        block_height: BlockHeight,
+    ) -> Self {
+        let inner_state = InnerSyncState::from_config(
+            min_connected_reserved_peers,
+            time_until_synced,
+            block_height,
+        );
+        let timer = DeadlineClock::new();
+
+        let initial_sync_state =
+            SyncState::from_config(min_connected_reserved_peers, time_until_synced);
+
+        let (state_sender, state_receiver) =
+            tokio::sync::watch::channel(initial_sync_state);
+
+        Self {
+            peer_connections_stream,
+            min_connected_reserved_peers,
+            time_until_synced,
+            block_stream,
+            state_sender,
+            state_receiver,
+            inner_state,
+            timer,
+        }
+    }
+
+    async fn update_timeout(&mut self) {
+        if self.inner_state.is_sufficient() {
+            self.timer
+                .set_timeout(self.time_until_synced, OnConflict::Overwrite)
+                .await;
+        } else {
+            self.timer.clear().await;
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl RunnableService for SyncTask {
+    const NAME: &'static str = "fuel-core-consensus/poa/sync-task";
+
+    type SharedData = watch::Receiver<SyncState>;
+    type TaskParams = ();
+
+    type Task = SyncTask;
+
+    fn shared_data(&self) -> Self::SharedData {
+        self.state_receiver.clone()
+    }
+
+    async fn into_task(
+        self,
+        _: &StateWatcher,
+        _: Self::TaskParams,
+    ) -> anyhow::Result<Self::Task> {
+        Ok(self)
+    }
+}
+
+#[async_trait::async_trait]
+impl RunnableTask for SyncTask {
+    #[tracing::instrument(level = "debug", skip_all, err, ret)]
+    async fn run(&mut self, watcher: &mut StateWatcher) -> anyhow::Result<bool> {
+        let mut should_continue = true;
+
+        tokio::select! {
+            _ = watcher.while_started() => {
+                should_continue = false;
+            }
+            latest_count = self.peer_connections_stream.next() => {
+                if let Some(latest_count) = latest_count {
+                    if self.state_sender.send_if_modified(|sync_state: &mut SyncState| {
+                        if self.inner_state.change_state_on_peers_update(latest_count, self.min_connected_reserved_peers) {
+                            *sync_state = self.inner_state.sync_state();
+                            true
+                        } else {
+                            false
+                        }
+                    }) {
+                        self.update_timeout().await;
+                    }
+                }
+
+            }
+            block = self.block_stream.next() => {
```

Comment:
The same here:

<img width="1031" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/f70a3b5b-fe44-4527-a2d6-166a1f648684">

You can see that we can handle restarting of the timer in the case of `SufficientPeers` state.